tokio-console/src/view/durations.rs

```diff
@@ -0,0 +1,104 @@
+use std::cmp;
+
+use tui::{
+    layout::{self},
+    widgets::Widget,
+};
+
+use crate::{
+    state::histogram::DurationHistogram,
+    view::{self, mini_histogram::MiniHistogram, percentiles::Percentiles},
+};
+
+// This is calculated so that a legend like the below generally fits:
+// │0647.17µs  909.31µs │
+const MIN_HISTOGRAM_BLOCK_WIDTH: u16 = 22;
+
+/// This is a tui-rs widget to visualize durations as a list of percentiles
+/// and if possible, a mini-histogram too.
+///
+/// This widget wraps the [`Percentiles`] and [`MiniHistogram`] widgets which
+/// are displayed side by side. The mini-histogram will only be displayed if
+///   a) UTF-8 support is enabled via [`Styles`]
+///   b) There is at least a minimum width left after drawing the percentiles
```

Comment:
nit, take it or leave it: maybe we should mention what the minimum width is?

Reply:
Added.